src/components/Modals/FiatRamps/FiatRamps.tsx

```diff
@@ -1,24 +1,69 @@
 import { Modal, ModalCloseButton, ModalContent, ModalOverlay } from '@chakra-ui/modal'
-import { MemoryRouter } from 'react-router-dom'
+import { MemoryRouter, useLocation } from 'react-router-dom'
 import { useModal } from 'hooks/useModal/useModal'
 
 import { FiatRampsRoutes } from './FiatRampsCommon'
 import { FiatRampsRouter } from './FiatRampsRouter'
 
+import { useHistory } from 'react-router-dom'
+import { IconButton, ModalHeader } from '@chakra-ui/react'
+import { ArrowBackIcon } from '@chakra-ui/icons'
+
+
 const entries = [FiatRampsRoutes.Select, FiatRampsRoutes.Manager]
 
+
+const FiatRampsContent = () => {
+  const history = useHistory()
+  const location = useLocation()
+
+  if(location.pathname == FiatRampsRoutes.Select) {
+    return (
+      <>
+        <ModalCloseButton />
+        <FiatRampsRouter />
+      </>
+    )
+  }
+
+  return (
+    <>
+      <ModalHeader>
+        <IconButton 
+          variant="ghost"
+          icon={<ArrowBackIcon />}
+          onClick={() => {
+            history.goBack()
+          }}
```

Comment:
You can remove that wrapping arrow function to make it one less function in the stack trace

```suggestion
          onClick={history.goBack}
```